src/Sulu/Component/Rest/ListBuilder/Metadata/ProviderInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Metadata;
+
+use Metadata\ClassMetadata;
+
+/**
+ * Interface for metadata-providers.
+ */
+interface ProviderInterface
```

Comment:
Maybe explain here more about what a Provider does.

i.e. that each provider represents an "extension" in the list component and may/does have its own Metadata\Factory. etc. etc.